posthog/models/history_logging.py

```diff
@@ -0,0 +1,196 @@
+import dataclasses
+from itertools import zip_longest
+from typing import Dict, Iterable, List, Optional, Tuple, Union
+
+from rest_framework import serializers
+
+from posthog.models import HistoricalVersion, User
+
+
+class HistoryListItemSerializer(serializers.Serializer):
+    email = serializers.EmailField(read_only=True)
+    name = serializers.CharField(read_only=True)
+    user_id = serializers.IntegerField(read_only=True)
+    action = serializers.CharField(read_only=True)
+    detail = serializers.DictField(read_only=True)
+    created_at = serializers.CharField(read_only=True)
+
+
+@dataclasses.dataclass(frozen=True)
+class HistoryListItem:
+    email: Optional[str]
+    name: Optional[str]
+    action: str
+    detail: Dict[str, Union[int, str, Dict]]
```

Comment:
Can we add a bit more detail here type wise? I think we do actually know that we have an `id` and `key`?

Reply:
Can we also make this an array, one for each key changed? That seems a bit more developer friendly.